Tools/ViewsEstimators.py

```diff
@@ -6,118 +6,333 @@
+            ):
     
+        """ Fit the model based on the indicator values. """
+        if isinstance(X, np.ndarray):
+            X = pd.DataFrame(X)
+        if self.split_by not in X.columns:
+            raise ValueError(f"split_by column '{self.split_by}' not found in X")
+
+        split_indicator = X[self.split_by].astype(int)
+        X = X.drop(columns=[self.split_by])
+        X, y = check_X_y(X, y, accept_sparse=False, force_all_finite='allow-nan', dtype=np.float64)
 
+        self.ones_ = self._resolve_estimator(self.ones_model_name)
+        if self.ones_model_params:
+            self.ones_.set_params(**self.ones_model_params)
+        self.ones_.fit(X[split_indicator == 1], y[split_indicator == 1])
 
+        self.zeros_ = self._resolve_estimator(self.zeros_model_name)
+        if self.zeros_model_params:
+            self.zeros_.set_params(**self.zeros_model_params)
+        self.zeros_.fit(X[split_indicator == 0], y[split_indicator == 0])
+
+        self.is_fitted_ = True
+        return self
+
+    def predict(self, X: Union[np.ndarray, pd.DataFrame]):
+        """ Predict the response based on the split_by indicator values. """
+        if isinstance(X, np.ndarray):
+            X = pd.DataFrame(X)
+
+        check_is_fitted(self, 'is_fitted_')
+
+        if self.split_by not in X:
+            raise ValueError(f"split_by column '{self.split_by}' not found in X")
+
+        split_indicator = X[self.split_by].astype(int)
+        X = X.drop(columns=[self.split_by])
+
+        pred_ones = self.ones_.predict(X[split_indicator == 1])
+        pred_zeros = self.zeros_.predict(X[split_indicator == 0])
+
+        pred = np.zeros(X.shape[0])
+        pred[split_indicator == 1] = pred_ones
+        pred[split_indicator == 0] = pred_zeros
+
+      # Create a DataFrame with predictions and split_by values
+        result = pd.DataFrame({
+            'prediction': pred,
+            'split_by': split_indicator
+        })
+
+        return result
+
+def test_fixed_first_split_regression(zeros_model_name:str='linear', 
+                                      ones_model_name:str='linear', 
+                                      split_by = "split_by"):
+    """ Validate estimator using sklearn's provided utility and ensure it can fit and predict on fake dataset. """
+    # Create a regression dataset
+    X, y = make_classification(n_samples=100, n_features=20)
+
+    # Convert to DataFrame and add a 'split_by' column
+    X = pd.DataFrame(X, columns=[f'feature_{i}' for i in range(X.shape[1])])
+    X['split_by'] = np.random.randint(0, 2, len(y))  # Random 0s and 1s for the indicator
+
+    # Initialize and fit the FixedFirstSplitRegression model
+    reg = FixedFirstSplitRegression(zeros_model_name=zeros_model_name, 
+                                    ones_model_name=ones_model_name, 
+                                    split_by=split_by)
+
+    reg.fit(X, y)
 
-#if __name__ == '__main__':
```

Comment:
Why remove?

Reply:
I do not see why we need it. Do we plan to execite this file directly? My understanding is that we will import a class from its file in some other file like for example this `from Tools.models.hurdle_regression_model import HurdleRegression`, then we do not need this line. Afaik, this line is needed only if the code in the file needs to be executed directly. Let me know if you think it is needed.